src/main/java/com/myorg/mvc/controllers/UserController.java

```diff
@@ -1,40 +1,81 @@
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 
 import java.util.List;
 
 @Controller
-@RequestMapping("/user")
 public class UserController {
+    @Autowired
+    @Qualifier("UserService")
+    private UserService userService;
 
-    private final UserService userService;
+    @GetMapping
+    public String showHome() {
+        return "index";
+    }
 
-    @Autowired
-    public UserController(UserService userService) {
-        this.userService = userService;
+    @GetMapping("/login")
+    public String logIn() {
+        return "login";
     }
 
-    @RequestMapping("/all")
-    public List getUsers() {
-        return userService.getAllUsers();
+    @GetMapping("/register_success")
+    public String success() {
+        return "register_success";
     }
 
-    @PostMapping
-    public void addNewUser(@RequestBody User user) {
+    @GetMapping("/registration_form")
+    public String showRegistrationForm(Model model) {
+        User user = userService.createEmptyUser();
+        model.addAttribute("user", user);
+        return "registration_form";
+    }
+
+    @PostMapping("/users")
+    public String saveUser(@ModelAttribute User user) {
         userService.addNewUser(user);
+        return "redirect:/register_success";
     }
 
-    @DeleteMapping("{userId}")
-    public void deleteUser(@PathVariable("userId") Long userID) {
+    @GetMapping("/users")
+    public String getUsers(Model model) {
+        List<User> allUsers = userService.getAllUsers();
+        model.addAttribute("users", allUsers);
+        return "all";
+    }
+
+    @GetMapping("/users/{id}")
+    public String getByID(@PathVariable Long id, Model model) {
+        User userById = userService.getUserById(id);
+        model.addAttribute("userById", userById);
+        return "user_by_id";
+    }
+
+    @DeleteMapping("/users/{id}")
+    public String deleteUser(@PathVariable("id") Long userID) {
         userService.deleteUser(userID);
+        return "redirect:/users";
+    }
+
+    @PutMapping("/{userId}&{firstName}&{lastName}&{email}&{birthday}")
+    public String updateUser(@PathVariable("userId") Long userId,
+                             @PathVariable("firstName") String firstName,
+                             @PathVariable("lastName") String lastName,
+                             @PathVariable("email") String email,
+                             @PathVariable("birthday") String birthday, Model model) {
```

Comment:
Is it possible to use object here ? Because sending firstName emails address is not secure keep them in url.